Actions/CheckForNewWarnings/CheckForNewWarnings.ps1

```diff
@@ -0,0 +1,49 @@
+Param(
+    [string] $token,
+    [string] $project,
+    [object] $settings,
+    [string] $targetBranch,
+    [string] $prBuildOutputFile
+)
+
+if (-not $ENV:GITHUB_BASE_REF)
+{
+    Write-Host "Checking for warnings only runs on pull requests."
+    return
+}
+
+if (-not $settings.failOnNewWarnings)
+{
+    Write-Host "Failing on new warnings is disabled in the settings."
+    return
+}
+
+try
+{
+    Write-Host "::group::Analyzing build for new warnings..."
+
+    Import-Module (Join-Path $PSScriptRoot "..\Github-Helper.psm1" -Resolve) -DisableNameChecking
+    Import-Module (Join-Path $PSScriptRoot ".\CheckForWarningsUtils.psm1" -Resolve) -DisableNameChecking
+
+    $baselineWorkflowRunId,$baselineWorkflowSHA = FindLatestSuccessfulCICDRun -repository $env:GITHUB_REPOSITORY -branch $targetBranch -token $token -retention $settings.incrementalBuilds.retentionDays
+    if ($project) { $projectName = $project } else { $projectName = $env:GITHUB_REPOSITORY -replace '.+/' }
+
+    $mask = Get-Item $prBuildOutputFile | Select-Object -ExpandProperty BaseName
```

Comment:
The _buildMode_ should be included in the mask as well.

There's actually an action, "Calculate Artifact names" that calculates the all the artifact names for a project build.
In this case it is a coincidence that it's called the same way as the build output file itself. The proper way to get the build output artifact name is from that action (which currently runs after _RunPipeline_, so it's not trivial).

[Here](https://github.com/microsoft/AL-Go/blob/7d7d46b728ac7131f031d0765efba6a1ad025383/Actions/CalculateArtifactNames/CalculateArtifactNames.ps1#L52) you can check how artifact names are constructed.